enhancements/network/system-networkpolicies.md

```diff
@@ -0,0 +1,386 @@
+     the restrictive ones).
+
+       a. The permissive NetworkPolicies could be created by the
+          individual operators along with the restrictive policies.
+
+       b. Alternatively, all of the permissive NetworkPolicies could
+          be created by the Cluster Network Operator.
+
+Option 2 is preferable to Option 1 because in general, NetworkPolicy
+is easier to work with when namespaces always have either a "default
+deny" or "allow to all" baseline policy. (See Appendix for gory
+details.)
+
+Of the two sub-options, 2b ("CNO creates permissive policies") seems
+nice because it means that individual operators wouldn't need to track
+the restricted-vs-open option; they'd just always configure themselves
+as restricted, and CNO would fix things up if they were supposed to be
+open.
+
+However, 2b would make upgrades more complicated, since we have to
+ensure that no component ever creates a restrictive NetworkPolicy
+unless CNO has already create a permissive NetworkPolicy (if one is
+needed). The easiest way to do that would be to add the "permissive
+NetworkPolicy" feature to CNO one full Y-stream release before adding
+the restrictive NetworkPolicies to other operators...
+
+#### OpenShift SDN Multitenant Mode
+
+OCP 4.x still supports the old pre-NetworkPolicy "Multitenant" mode
+where all namespaces are isolated from each other by default. This
+doesn't really work as well in OCP 4.x as it did in OCP 3.x, since
+there are now many more system namespaces that need to talk to each
+other.
+
+In Multitenant mode, it is not possible to fine-tune permissions; any
+namespace which _either_ needs to be accessible from arbitrary other
+namespaces, _or_ needs to be able access arbitrary other namespaces,
+must be joined to the "default" namespace. In 4.7.0, this set of
+namespaces is:
+
+- `openshift-dns`
+- `openshift-image-registry`
+- `openshift-ingress`
+- `openshift-kube-apiserver`
+- `openshift-monitoring`
+- `openshift-operator-lifecycle-manager`
+- `openshift-user-workload-monitoring`
+
+We also create a second "control-plane" set of joined namespaces,
+which are all able to access each other, but which remain isolated
+from other (non-default) namespaces:
+
+- `kube-system`
+- `openshift-ansible-service-broker`
+- `openshift-apiserver`
+- `openshift-authentication`
+- `openshift-authentication-operator`
+- `openshift-etcd`
+- `openshift-etcd-operator`
+- `openshift-service-catalog-apiserver`
+- `openshift-service-catalog-controller-manager`
+- `openshift-template-service-broker`
+
+(The presence of `kube-system` there is probably a mistake, since
+there are no pods in `kube-system`...)
+
+These lists occasionally need to be adjusted. It would be good if we
+could find a way to figure these out automatically, either based on
+analyzing the NetworkPolicies that the operators create, or by having
+the operators annotate their namespaces with hints to openshift-sdn.
+
+### Risks and Mitigations
```

Comment:
The biggest risk, though, is that we block something unexpected not covered by e2e tests. Something, for example, shipped via OLM.